sidechain/consensus/common/src/header_db.rs

```diff
@@ -0,0 +1,88 @@
+/*
+	Copyright 2021 Integritee AG and Supercomputing Systems AG
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+
+*/
+use itp_types::H256;
+use its_primitives::{traits::Header as HeaderT, types::header::SidechainHeader};
+use std::{collections::HashMap, hash::Hash as HashT};
+
+// Normally implemented on the Client in substrate
+pub trait HeaderDbTrait {
+	type Header: HeaderT;
+	/// Retrieves Header for the corresponding block hash.
```

Comment:
I think that that the trait should have a doc comment as well, not just the method.